algosdk/abi.py

```diff
@@ -0,0 +1,570 @@
+        return True
+
+    @staticmethod
+    def make_uint_type(type_size):
+        if (
+            not isinstance(type_size, int)
+            or type_size % 8 != 0
+            or type_size < 8
+            or type_size > 512
+        ):
+            raise error.ABITypeError(f"unsupported uint bitSize: {type_size}")
+        return Type(BaseType.Uint, bit_size=type_size)
+
+    @staticmethod
+    def make_byte_type():
+        return Type(BaseType.Byte)
+
+    @staticmethod
+    def make_ufixed_type(type_size, type_precision):
+        if (
+            not isinstance(type_size, int)
+            or type_size % 8 != 0
+            or type_size < 8
+            or type_size > 512
+        ):
+            raise error.ABITypeError(
+                f"unsupported ufixed bitSize: {type_size}"
+            )
+        if (
+            not isinstance(type_precision, int)
+            or type_precision > 160
+            or type_precision < 1
+        ):
+            raise error.ABITypeError(
+                f"unsupported ufixed precision: {type_precision}"
+            )
+        return Type(
+            BaseType.Ufixed, bit_size=type_size, precision=type_precision
+        )
+
+    @staticmethod
+    def make_bool_type():
+        return Type(BaseType.Bool)
+
+    @staticmethod
+    def make_static_array_type(arg_type, array_len):
+        assert isinstance(arg_type, Type)
+        return Type(
+            BaseType.ArrayStatic,
+            child_type=arg_type,
+            child_types=list(),
+            static_length=array_len,
+        )
+
+    @staticmethod
+    def make_address_type():
+        return Type(BaseType.Address)
+
+    @staticmethod
+    def make_dynamic_array_type(arg_type):
+        assert isinstance(arg_type, Type)
+        return Type(
+            BaseType.ArrayDynamic, child_type=arg_type, child_types=list()
+        )
+
+    @staticmethod
+    def make_string_type():
+        return Type(BaseType.String)
+
+    @staticmethod
+    def make_tuple_type(arg_types):
+        if len(arg_types) >= 2 ** 16:
+            raise error.ABITypeError(f"tuple args exceed 2^16")
+        assert isinstance(arg_types, list)
+        return Type(
+            BaseType.Tuple, child_types=arg_types, static_length=len(arg_types)
+        )
+
+    @staticmethod
+    def type_from_string(s):
+        if s.endswith("[]"):
+            array_arg_type = Type.type_from_string(s[:-2])
+            return Type.make_dynamic_array_type(array_arg_type)
+        elif s.endswith("]"):
+            static_array_regex = "^([a-z\d\[\](),]+)\[([1-9][\d]*)]$"
+            matches = re.search(static_array_regex, s)
+            try:
+                static_length = int(matches.group(2))
+                array_type = Type.type_from_string(matches.group(1))
+                return Type.make_static_array_type(array_type, static_length)
+            except:
+                raise error.ABITypeError(f"malformed static array string: {s}")
```

Comment:
```suggestion
            except Exception as e:
                raise error.ABITypeError(f"malformed static array string: {s}") from e
```
Since you are raising an error only as a result of another error, it's a good idea to use `raise ... from ...` to preserve this relationship. More info here: https://stefan.sofa-rockers.org/2020/10/28/raise-from/

There are other places below that would benefit from this as well.